src/main/java/org/joda/beans/ser/bin/AbstractBinReader.java

```diff
@@ -322,11 +322,13 @@ Object parseSimple(int typeByte, Class<?> type) throws Exception {
                 // handle case where property type has changed from integral to float
                 return Float.valueOf((float) value);
 
-            } else {
+            } else if (type == Integer.class || type == int.class) {
                 if (value < Integer.MIN_VALUE || value > Integer.MAX_VALUE) {
                     throw new IllegalArgumentException("Invalid binary data: Expected int, but was " + value);
                 }
                 return Integer.valueOf((int) value);
+            } else {
+                return Long.valueOf(value);
```

Comment:
I've changed the bin reader too for consistency, even though I don't think it suffers from the same issue as we don't ever serialize to binary format without type information.